File: chapter_one/binarySearch.js
const binarySearch = (arr, target) => {
    let low = 0
    let high = arr.length - 1

    while ( low <= high ) {
        let middle = Math.floor( (low + high) / 2)
        let midNum = arr[middle]

        if ( midNum === target ) {
            return middle
        }
        else if ( midNum < target ) {
            low = middle + 1
        }
        else {
            high = middle - 1
        }
    }

    return -1
}

let testArr = []

for ( let i = 0; i < 1000; i += 1 ) {
    let randomNum = Math.floor((Math.random() * 100))
    testArr.push(randomNum)
}
testArr.sort((a,b) => a - b)

console.log(testArr, binarySearch(testArr, 5))

File: chapter_one/binarySearch.py
import random


def binary_search(arr, target):
    low = 0
    high = len(arr) - 1

    while low <= high:
        middle = (low + high) // 2
        if arr[middle] == target:
            return middle
        elif arr[middle] < target:
            low = middle + 1
        else:
            high = middle - 1

    return -1


test_arr = [random.randint(0, 100) for i in range(100)]
test_arr.sort()

print(test_arr, binary_search(test_arr, 5))
